src/ffi.rs

```diff
@@ -354,21 +370,25 @@ pub extern "C" fn threshold_keygen(n: usize, t: usize, seed: &[u8], keys: *mut K
         n,
     };
 
-    unsafe { *keys = keys_local };
+    unsafe {
+        *keys = Box::into_raw(Box::new(keys_local));
+    };
 }
 
 /// Generates a single private key from the provided seed.
 ///
+/// The return value should be destroy with `destroy_keypair`.
+///
 /// # Safety
 ///
 /// The seed MUST be at least 32 bytes long
 #[no_mangle]
-pub extern "C" fn keygen(seed: *const Buffer, keypair: *mut Keypair) {
+pub extern "C" fn keygen(seed: *const Buffer, keypair: *mut *mut Keypair) {
     let seed = <&[u8]>::from(unsafe { &*seed });
     let mut rng = get_rng(&seed);
     let (private, public) = BlindThresholdSigs::keypair(&mut rng);
     let keypair_local = Keypair { private, public };
-    unsafe { *keypair = keypair_local };
+    unsafe { *keypair = Box::into_raw(Box::new(keypair_local)) };
```

Comment:
Do you think that using `std::ptr::write(keypair, keypair_local)` is a better option than the `*mut *mut` pattern? Not sure what the canonical way to do this is. 

The current way of `*ptr = value` was definitely bad due to it calling drop on potentially uninitialized memory.

Reply:
I think that we can't work with the "single pointer" approach anyway, because otherwise the caller has to allocate enough memory, which seems like a greater hassle than the *mut *mut approach. Wdyt?